metrics/metrics.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2015 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package metrics
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/cactus/go-statsd-client/statsd"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/streadway/amqp"
+
+	"github.com/letsencrypt/boulder/rpc"
+)
+
+// HTTPMonitor stores some server state
+type HTTPMonitor struct {
+	stats           statsd.Statter
+	statsPrefix     string
+	handler         http.Handler
+	openConnections *int64
+}
+
+// NewHTTPMonitor returns a new initialized HTTPMonitor
+func NewHTTPMonitor(stats statsd.Statter, handler http.Handler, prefix string) HTTPMonitor {
+	conns := int64(0)
+	return HTTPMonitor{stats: stats, handler: handler, statsPrefix: prefix, openConnections: &conns}
+}
+
+// Handle wraps handlers and records various metrics about requests to these handlers
+// and sends them to StatsD
+func (h *HTTPMonitor) Handle() http.Handler {
+	return http.HandlerFunc(h.watchAndServe)
+}
+
+func (h *HTTPMonitor) watchAndServe(w http.ResponseWriter, r *http.Request) {
+	h.stats.Inc(fmt.Sprintf("%s.HTTP.Rate", h.statsPrefix), 1, 1.0)
+	open := atomic.AddInt64(h.openConnections, 1)
```

Comment:
This is the number of in-flight requests, not open connections.

To get open connections, you have to dig into the Server.ConnState stuff.